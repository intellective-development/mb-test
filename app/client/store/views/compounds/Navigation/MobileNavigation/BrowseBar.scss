@import "~client/store/views/style/variables";

$browse-bar-height: 34px;
$search-disclose-animation-duration: 200ms;

.cmMBrowseBar {
  display: flex;
  height: $browse-bar-height;
}

.cmMBrowseBar_LeftWrapper {
  display: flex;
  flex: none;

  // animation
  transform: translateZ(0); // iOS performance hack
  transition: all $search-disclose-animation-duration ease-in-out;
  max-width: 150px; // roughly the right size
  opacity: 1;
}
.cmMBrowseBar_LeftWrapper__SearchActive {
  max-width: 0;
  opacity: 0;
}

.cmMBrowseBar_Left {
  transform: translateZ(0); // iOS performance hack
  display: flex;
  align-items: center;
  padding-right: $standard-gutter;
  border-right: $default-border;
  padding-left: $standard-gutter;
  transition: all 250ms ease-in-out;
}
.cmMBrowseBar_Left__Active {
  background-color: $brand-pale-blue;
}

.cmMBrowseBar_Right {
  display: flex;
  align-items: center;
  flex: auto;
  padding-right: $standard-gutter;
  margin-left: 12px;
  overflow-x: hidden;
}

.cmMBrowseBar_CategoryPrompt {
  font-weight: $heavy-font-weight;
  margin-top: 1px;
}
.cmMBrowseBar_DicloseIcon {
  margin-left: 7px;
  display: inline-block;
  transition: transform 120ms ease-in-out;
}
.cmMBrowseBar_DicloseIcon__Active { transform: rotate(180deg); }


$search-icon-spacing: 36px;

.cmMBrowseBar_Search_Wrapper {
  display: flex;
  position: relative;
  flex: auto;
}
.cmMBrowseBar_SearchIcon {
  position: absolute;
  left: 0;
  top: 3px;

  // ensures it won't block hover/focus events for the input it is sitting on top of
  pointer-events: none;
}

input.cmMBrowseBar_SearchInput { // input used to override system styles
  @include font-size(14);
  flex: auto;
  height: $browse-bar-height;
  color: $brand-black;
  border: none;
  padding: 0;
  color: $brand-black;
  white-space: nowrap;
  text-overflow: ellipsis;
  text-transform: capitalize;
  padding-left: $search-icon-spacing;

  &::placeholder {
    text-transform: initial;
    color: $brand-medium-grey;
  }
}

.cmMBrowseBar_ClearSearch {
  cursor: pointer;
  backface-visibility: hidden; // prevents clipping with opacity
  padding: 0 8px 2px;
  margin-right: 8px;
  transition: all 120ms ease-in-out;
  opacity: 0.75;

  &:hover {
    opacity: 1;
  }
}
.cmMBrowseBar_ClearSearch__SearchEmpty {
  visibility: hidden;
  opacity: 0;
}
.cmMBrowseBar_ClearSearch__SearchActive {
  display: none;
}
.cmMBrowseBar_CancelSearch {
  @include font-size(14);

  font-weight: $heavy-font-weight;
  color: $brand-red;
  display: block;
  transition: all $search-disclose-animation-duration ease-in-out;
  transform: translateY(5px);
  opacity: 0;
  padding-top: 2px;
}
.cmMBrowseBar_CancelSearch__SearchActive {
  opacity: 1;
  transform: translateY(0);
}
