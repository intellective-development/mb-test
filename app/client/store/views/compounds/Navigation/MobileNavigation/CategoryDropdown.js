// @flow
import * as React from 'react';
import _ from 'lodash';
import bindClassNames from 'shared/utils/bind_classnames';
import type { ContentLayout } from 'store/business/content_layout';
import type { ImageContent } from 'store/business/content_module';
import { NAVIGATION_CATEGORY_DEFAULTS } from '../constants';

import withContentLayout from '../../../compounds/ContentLayout';
import { MBLink, MBText, withUniqueId } from '../../../elements';
import styles from './CategoryDropdown.scss';

const cn = bindClassNames(styles);

type CategoryDropdownProps = {|
  is_showing: boolean,
  content_layout: ContentLayout
|};
export const CategoryDropdown = ({is_showing, content_layout}: CategoryDropdownProps) => {
  let category_links = NAVIGATION_CATEGORY_DEFAULTS;
  if (content_layout){
    // we assume that this content layout will only have one element, which contains the desired links
    category_links = _.get(content_layout, 'content[0].config.links');
  }

  return (
    <div className={cn('cmMCategoryDropdown_Wrapper', {cmMCategoryDropdown_Wrapper__DropdownVisible: is_showing})}>
      <ul className={cn('cmMCategoryDropdown', {cmMCategoryDropdown__Visible: is_showing})}>
        {category_links.map(el => (
          <CategoryElement
            key={el.name}
            name={el.name}
            icon_banner={el.icon_banner}
            action_url={el.action_url}
            disabled={!el.action_url} />
        ))}
      </ul>
    </div>
  );
};

type CategoryElementProps = {|
  action_url: string,
  name: 'wine' | 'beer' | 'liquor' | 'mixers' | 'gifts',
  icon_banner?: ImageContent,
  disabled?: boolean
|};
const CategoryElement = ({ action_url, name, icon_banner, disabled }: CategoryElementProps) => {
  return (
    <li>
      <MBLink.View
        disabled={disabled}
        className={cn('cmMCategoryDropdown_Element', {cmMCategoryDropdown_Element__Disabled: disabled})}
        href={action_url}>
        <CategoryElementIcon category_name={name} icon_banner={icon_banner} />
        <MBText.Span className={styles.cmMCategoryDropdown_ElementName}>{name}</MBText.Span>
        {disabled && <MBText.Span className={styles.cmMCategoryDropdown_UnavailableMessage}>&ensp;(not available)</MBText.Span>}
      </MBLink.View>
    </li>
  );
};

const CategoryElementIcon = ({category_name, icon_banner}) => {
  if (!icon_banner) return null;

  return (
    <img
      className={styles.cmMCategoryDropdown_ElementIcon}
      alt={category_name}
      src={icon_banner.image_url}
      height={icon_banner.image_height}
      width={icon_banner.image_width} />
  );
};

export default withUniqueId('content_layout_id')(withContentLayout('Web_Navigation_Mobile_Category')(CategoryDropdown));
