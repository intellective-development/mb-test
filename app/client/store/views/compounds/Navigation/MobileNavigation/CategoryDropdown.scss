@import "~client/store/views/style/variables";

$dropdown_height: 360px; // hardcoded, used to animate the dropdown
$dropdown_shadow_height: 4px;

// NOTE: that these classes are very, very similar to their counterparts in DesktopNavigation
.cmMCategoryDropdown_Wrapper {
  position: absolute;
  width: 100vw;
  left: 0;
  margin: 0;
  top: 100%; // stick it below the bottom of the container
  z-index: 10;
  border-top: $default-border;

  // this padding allows the dropdown's box shadow to show, despite the above overflow: hidden
  padding-bottom: $dropdown_shadow_height;

  overflow: hidden;
  visibility: hidden;
}
.cmMCategoryDropdown_Wrapper__DropdownVisible {
  visibility: visible;
}

.cmMCategoryDropdown {
  background: $brand-white;
  transition: all 200ms ease-in-out;
  margin-bottom: 0;
  padding: 12px 0 16px;
  box-shadow: 0 $dropdown_shadow_height/2  $dropdown_shadow_height 0 rgba(0, 0, 0, 0.33);

  transition: all 250ms ease-in; // animate out
  transform: translateY(-$dropdown_height);
  opacity: 0.5;
}
.cmMCategoryDropdown__Visible {
  transition-timing-function: ease-out; // animate in
  transform: translateY(0);
  opacity: 1;
}

.cmMCategoryDropdown_Element {
  @include font-size(24);

  display: inline-block;
  padding: 13px 24px;
  color: $brand-black;
}

.cmMCategoryDropdown_Element__Disabled {
  opacity: 0.25;
}

.cmMCategoryDropdown_ElementIcon {
  margin-right: 24px;
}

.cmMCategoryDropdown_ElementName {
  display: inline-block;
  padding-top: 4px;
  text-transform: capitalize;
}

.cmMCategoryDropdown_UnavailableMessage {
  @include font-size(13);
}