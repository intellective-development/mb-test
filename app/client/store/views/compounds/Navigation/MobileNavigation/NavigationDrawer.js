// @flow

import * as React from 'react';
import { connect } from 'react-redux';
import * as Ent from '@minibar/store-business/src/utils/ent';
import { ui_actions } from 'store/business/ui';
import { user_selectors } from 'store/business/user';
import type { User } from 'store/business/user';

import DeliveryInfo from '../shared/DeliveryInfo';
import { MBAppStoreLink, MBDrawer, MBDynamicIcon, MBLink, MBText, MBTouchable } from '../../../elements';
import styles from './NavigationDrawer.scss';

type NavigationDrawerProps = {
  show: boolean,
  closeDrawer: () => void,

  // Connect Props
  user: User,
  showDeliveryInfoModal: () => void,
  showHelpModal: () => void
};
const NavigationDrawer = ({show, closeDrawer, user, showDeliveryInfoModal, showHelpModal}: NavigationDrawerProps) => {
  return (
    <MBDrawer
      open={show}
      closeDrawer={closeDrawer}
      className={styles.cmMNavDrawer_Container}>
      <div className={styles.cmMNavDrawer_Content}>
        <CloseLink handleClick={closeDrawer} />
        <DeliveryLink showDeliveryInfoModal={showDeliveryInfoModal} closeDrawer={closeDrawer} />
        <NavigationLinks user={user} closeDrawer={closeDrawer} showHelpModal={showHelpModal} />
        <MBAppStoreLink className={styles.cmMNavDrawer_AppStoreLink} />
      </div>
    </MBDrawer>
  );
};

const CloseLink = ({handleClick}) => (
  <MBTouchable onClick={handleClick}>
    <MBDynamicIcon name="x_close" width={30} height={30} className={styles.cmMNavDrawer_CloseLink} />
  </MBTouchable>
);

const DeliveryLink = ({showDeliveryInfoModal, closeDrawer}) => (
  <MBTouchable
    role="button"
    className={styles.cmMNavDrawer_DeliveryLink}
    onClick={() => {
      showDeliveryInfoModal();
      closeDrawer();
    }} >
    <DeliveryInfo />
    <MBDynamicIcon name="chevron_right" width={28} height={28} />
  </MBTouchable>
);

const NavigationLinks = ({user, closeDrawer, showHelpModal}) => (
  <ul className={styles.cmMNavDrawer_ItemLinkContainer}>
    <li>
      <MBLink.Text
        href="/store/"
        beforeNavigate={closeDrawer}
        className={styles.cmMNavDrawer_ItemLink}
        standard={false}>
        Shop
      </MBLink.Text>
    </li>
    <li>
      <MBLink.Text
        href="/referrals"
        className={styles.cmMNavDrawer_ItemLink}
        native_behavior
        standard={false}>
        Share & Save $10
      </MBLink.Text>
    </li>
    <li>
      <MBLink.Text
        href="/account/orders"
        className={styles.cmMNavDrawer_ItemLink}
        native_behavior
        standard={false}>
        Orders
      </MBLink.Text>
    </li>
    <li>
      <MBLink.Text
        href="/account/overview"
        className={styles.cmMNavDrawer_ItemLink}
        native_behavior
        standard={false}>
        Account
        <MBText.Span className={styles.cmMNavDrawer_UserName}>
          &emsp;{user && `${user.first_name} ${user.last_name.slice(0, 1)}.`}
        </MBText.Span>
      </MBLink.Text>
    </li>
    <li>
      <MBTouchable
        className={styles.cmMNavDrawer_ItemLink}
        onClick={() => {
          closeDrawer();
          showHelpModal();
        }} >
        <MBText.Span>Help</MBText.Span>
      </MBTouchable>
    </li>
  </ul>
);

const NavigationDrawerSTP = () => {
  const findUser = Ent.find('user');

  return (state) => ({user: findUser(state, user_selectors.currentUserId(state))});
};
const NavigationDrawerDTP = {
  showDeliveryInfoModal: ui_actions.showDeliveryInfoModal,
  showHelpModal: ui_actions.showHelpModal
};
const NavigationDrawerContainer = connect(NavigationDrawerSTP, NavigationDrawerDTP)(NavigationDrawer);

export default NavigationDrawerContainer;
