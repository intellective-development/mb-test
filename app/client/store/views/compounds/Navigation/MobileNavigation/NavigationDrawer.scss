@import "~client/store/views/style/variables";

.cmMNavDrawer_Container {
  background-color: $brand-white;
}

.cmMNavDrawer_Content {
  display: flex;
  flex-direction: column;
  height: 100%;
}

.cmMNavDrawer_CloseLink {
  flex: none;
  padding: 9px 12px;
}

.cmMNavDrawer_DeliveryLink {
  flex: none;
  padding: 20px 6px 20px 4px; // icon (leftmost element) already has padding on it
  background-color: $brand-pale-blue;
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.cmMNavDrawer_ItemLinkContainer {
  flex: auto;
  padding-top: 8px;
  margin: 0;
}

.cmMNavDrawer_ItemLink {
  @include font-size(24);
  font-weight: $medium-font-weight;
  padding: 14px 16px;
  display: inline-block;
}

.cmMNavDrawer_UserName {
  @include font-size(14);
  color: $brand-dark-grey;
}

.cmMNavDrawer_AppStoreLink {
  align-self: stretch;
  flex: none;
  padding: 0 $standard-gutter $standard-gutter;
  justify-content: space-around;
}
