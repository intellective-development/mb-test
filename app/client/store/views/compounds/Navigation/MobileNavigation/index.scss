@import "~client/store/views/style/variables";

.cmMNav_Container {
  z-index: 1;
  position: relative; // needed to make the box-shadow appear
  box-shadow: 0 0 2px 0 rgba(0, 0, 0, 0.67);
  background-color: $brand-white;

  @media only screen and (min-width: $medium-min-width) {
    display: none;
  }
}

.cmMNav_Top {
  display: flex;

  height: 49px;
  border-bottom: $default-border;
}

.cmMNav_Bottom {
  height: 34px;
}

.cmMNav_Top_Element {
  display: flex;
  align-items: center;
}

.cmMNav_Top_Element_Left,
.cmMNav_Top_Element_Right {
  flex: none;
  width: 64px;
}
.cmMNav_Top_Element_Left {
  justify-content: flex-start;
  padding-left: $standard-gutter / 2;
  padding-right: $standard-gutter;
}
.cmMNav_Top_Element_Right {
  justify-content: flex-end;
  padding-left: $standard-gutter;
  padding-right: $standard-gutter;
}
.cmMNav_Top_Element_Center {
  flex: auto;
  justify-content: center;
}

.cmMNav_CartIcon {
  margin-right: 4px;
}

.cmMNav_CartCount {
  position: absolute;
  top: 6px;
  right: 15px;
}
