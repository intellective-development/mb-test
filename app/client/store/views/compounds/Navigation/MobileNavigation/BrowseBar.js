// @flow
import * as React from 'react';
import bindClassNames from 'shared/utils/bind_classnames';
import connectToObservables from 'shared/components/higher_order/connect_observables';
import locationStream from 'legacy_store/router/location_stream';


import { MBClickOutside, MBDynamicIcon, MBIcon, MBInput, MBText, MBTouchable } from '../../../elements';
import Search from '../shared/Search';
import CategoryDropdown from './CategoryDropdown';
import styles from './BrowseBar.scss';

const cn = bindClassNames(styles);

type BrowseBarProps = {|
  location_fragment: string
|};
type BrowseBarState = {|
  search_active: boolean,
  category_dropdown_open: boolean
|};
class BrowseBar extends React.Component<BrowseBarProps, BrowseBarState> {
  state = {search_active: false, category_dropdown_open: false}

  componentWillReceiveProps(next_props: BrowseBarProps){
    if (this.props.location_fragment !== next_props.location_fragment){
      this.hideCategoryDropdown();
    }
  }

  openSearch = () => {
    this.setState({search_active: true});
  }

  closeSearch = () => {
    this.setState({search_active: false});
  }

  toggleCategoryDropdown = () => {
    this.setState(prev_state => ({category_dropdown_open: !prev_state.category_dropdown_open}));
  }

  hideCategoryDropdown = () => {
    this.setState({category_dropdown_open: false});
  }

  render(){
    const { search_active, category_dropdown_open } = this.state;

    return (
      <nav className={styles.cmMBrowseBar}>
        <Categories
          search_active={search_active}
          category_dropdown_open={category_dropdown_open}
          toggleCategoryDropdown={this.toggleCategoryDropdown}
          hideCategoryDropdown={this.hideCategoryDropdown} />
        <SearchInput
          openSearch={this.openSearch}
          closeSearch={this.closeSearch}
          search_active={search_active} />
      </nav>
    );
  }
}

const Categories = ({search_active, category_dropdown_open, toggleCategoryDropdown, hideCategoryDropdown}) => {
  return (
    <MBClickOutside handleClickOutside={hideCategoryDropdown} disableOnClickOutside={!category_dropdown_open}>
      <div className={cn('cmMBrowseBar_LeftWrapper', {cmMBrowseBar_LeftWrapper__SearchActive: search_active})}>
        <MBTouchable
          className={cn('cmMBrowseBar_Left', {cmMBrowseBar_Left__Active: category_dropdown_open})}
          onClick={toggleCategoryDropdown}>
          <MBText.Span className={styles.cmMBrowseBar_CategoryPrompt} >Categories</MBText.Span>
          <MBIcon name="down_arrow_red" className={cn('cmMBrowseBar_DicloseIcon', {cmMBrowseBar_DicloseIcon__Active: category_dropdown_open})} />
        </MBTouchable>
        <CategoryDropdown is_showing={category_dropdown_open} hide={hideCategoryDropdown} />
      </div>
    </MBClickOutside>
  );
};

type SearchInputProps = {
  search_active: boolean,
  openSearch: () => void,
  closeSearch: () => void
}

const SearchInput = ({search_active, openSearch, closeSearch}: SearchInputProps) => {
  return (
    <Search
      onClose={closeSearch}
      renderInput={({ value, clear, onKeyDown, onChange, onFocus, onBlur}) => (
        <div className={styles.cmMBrowseBar_Right}>
          <div className={styles.cmMBrowseBar_Search_Wrapper}>
            <MBDynamicIcon
              name="search"
              width={29}
              height={29}
              className={cn('cmMBrowseBar_SearchIcon', {cmMBrowseBar_SearchIcon__SearchActive: search_active})} />
            <MBInput.Input
              value={value}
              onChange={onChange}
              onKeyDown={onKeyDown}
              onFocus={(event) => {
                openSearch();
                onFocus(event);
              }}
              className={styles.cmMBrowseBar_SearchInput}
              placeholder="Search products" />
          </div>
          <MBTouchable
            onClick={clear}
            className={cn('cmMBrowseBar_ClearSearch', {
              cmMBrowseBar_ClearSearch__SearchEmpty: !value,
              cmMBrowseBar_ClearSearch__SearchActive: !search_active
            })}>
            <MBIcon name="clear" />
          </MBTouchable>
          <MBTouchable onClick={() => { onBlur(); closeSearch(); }}>
            <MBText.Span className={cn('cmMBrowseBar_CancelSearch', {cmMBrowseBar_CancelSearch__SearchActive: search_active})}>
              Cancel
            </MBText.Span>
          </MBTouchable>
        </div>
      )} />
  );
};

export default connectToObservables(BrowseBar, {
  location_fragment: locationStream.map((location) => location.fragment)
});
