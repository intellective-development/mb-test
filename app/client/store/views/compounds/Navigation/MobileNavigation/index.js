// @flow
import * as React from 'react';
import bindClassNames from 'shared/utils/bind_classnames';

import { MBIcon, MBDynamicIcon, MBLink, MBTouchable } from '../../../elements';
import CartCountIndicator from '../shared/CartCountIndicator';
import NavigationDrawer from './NavigationDrawer';
import BrowseBar from './BrowseBar';
import styles from './index.scss';

const cn = bindClassNames(styles);

type MobileNavigationProps = {|
  is_checking_out: boolean
|}
type MobileNavigationState = {show_drawer: boolean};
class MobileNavigation extends React.Component<MobileNavigationProps, MobileNavigationState> {
  state = {show_drawer: false}

  showDrawer = () => {
    this.setState({show_drawer: true});
  }
  hideDrawer = () => {
    this.setState({show_drawer: false});
  }

  render(){
    const { is_checking_out } = this.props;

    return (
      <header className={styles.cmMNav_Container}>
        <div className={styles.cmMNav_Top}>
          <MenuButton onClick={this.showDrawer} is_hidden={is_checking_out} />
          <Logo />
          <CartInfo is_hidden={is_checking_out} />
        </div>
        <BottomBar is_hidden={is_checking_out} />
        <NavigationDrawer
          closeDrawer={this.hideDrawer}
          openDrawer={this.showDrawer}
          show={this.state.show_drawer} />
      </header>
    );
  }
}

const MenuButton = ({onClick, is_hidden}) => {
  if (is_hidden) return null;

  return (
    <MBTouchable
      onClick={onClick}
      className={cn('cmMNav_Top_Element', 'cmMNav_Top_Element_Left')}>
      <MBDynamicIcon name="hamburger" width={32} height={32} />
    </MBTouchable>
  );
};

const Logo = () => {
  return (
    <div className={cn('cmMNav_Top_Element', 'cmMNav_Top_Element_Center')}>
      <MBLink.View alt="home" href="/store/">
        <MBIcon name="mobile.minibar_logo" width="76" />
      </MBLink.View>
    </div>
  );
};

const CartInfo = ({is_hidden}) => {
  if (is_hidden) return null;

  return (
    <MBLink.View
      className={cn('cmMNav_Top_Element', 'cmMNav_Top_Element_Right')}
      href="/store/cart">
      <MBDynamicIcon name="cart_mobile" width={32} height={32} className={styles.cmMNav_CartIcon} />
      <CartCountIndicator className={styles.cmMNav_CartCount} />
    </MBLink.View>
  );
};

const BottomBar = ({is_hidden}) => {
  if (is_hidden) return null;

  return (
    <div className={styles.cmMNav_Bottom}>
      <BrowseBar />
    </div>
  );
};

export default MobileNavigation;
