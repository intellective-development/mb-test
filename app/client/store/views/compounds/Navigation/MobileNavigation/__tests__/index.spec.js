
import React from 'react';

import TestProvider from 'store/views/__tests__/utils/TestProvider';
import MobileNavigation from '../index';

describe('MobileNavigation', () => {
  it('renders', () => {
    expect(render(
      <TestProvider>
        <MobileNavigation />
      </TestProvider>
    )).toMatchSnapshot();
  });
});
