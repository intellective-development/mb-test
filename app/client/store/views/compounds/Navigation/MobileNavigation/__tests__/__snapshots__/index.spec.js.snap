// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileNavigation renders 1`] = `
<header>
  <div>
    <div
      class="el-touchable cmMNav_Top_Element cmMNav_Top_Element_Left"
      role="button"
      tabindex="-1"
    >
      <div
        class="elMBDynamicIcon"
      >
        <svg
          class=""
          height="32"
          viewBox="0 0 32 32"
          width="32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="#222"
            fill-rule="nonzero"
          >
            <path
              d="M5 17h22v-2H5zM5 24h22v-2H5zM5 10h22V8H5z"
            />
          </g>
        </svg>
      </div>
    </div>
    <div
      class="cmMNav_Top_Element cmMNav_Top_Element_Center"
    >
      <a
        alt="home"
        href="/store/"
      >
        <img
          alt="mobile.minibar_logo"
          class="el-mbicon--mobile.minibar_logo"
          src="/assets/components/elements/mb-icon/mobile/minibar_logo.png"
          srcset="/assets/components/elements/mb-icon/mobile/minibar_logo@2x.png 2x, /assets/components/elements/mb-icon/mobile/minibar_logo@3x.png 3x"
          width="76"
        />
      </a>
    </div>
    <a
      class="cmMNav_Top_Element cmMNav_Top_Element_Right"
      href="/store/cart"
    >
      <div
        class="elMBDynamicIcon"
      >
        <svg
          class=""
          height="32"
          viewBox="0 0 32 32"
          width="32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="#222"
            fill-rule="evenodd"
          >
            <path
              d="M13.5 24.5a1.5 1.5 0 1 1-3 0 1.5 1.5 0 0 1 3 0zM23 23a1.5 1.5 0 1 1 0 3 1.5 1.5 0 0 1 0-3"
            />
            <path
              d="M6.549 6.944L9.664 20h15.864L29.5 9.801l-2.143-.079-3.233 8.334h-12.85L8.16 5H4.051L2.5 6.547l1.455 1.35.955-.953z"
              fill-rule="nonzero"
            />
            <path
              d="M23.5 14l-.748 2h-9.79l-.462-2h11zm1.511-4l-.748 2H11.985l-.462-2H25.01zm1.5 1.843V12h-.06l.06-.157z"
            />
          </g>
        </svg>
      </div>
      <span
        class="elMBText elMBText_ResetSpacing cmNavCartCount cmNavCartCount__Hidden"
      >
        0
      </span>
    </a>
  </div>
  <div>
    <nav>
      <div
        class="cmMBrowseBar_LeftWrapper"
      >
        <div
          class="el-touchable cmMBrowseBar_Left"
          role="button"
          tabindex="-1"
        >
          <span
            class="elMBText elMBText_ResetSpacing"
          >
            Categories
          </span>
          <img
            alt="down_arrow_red"
            class="el-mbicon--down_arrow_red cmMBrowseBar_DicloseIcon"
            src="/assets/components/elements/mb-icon/down_arrow_red.png"
            srcset="/assets/components/elements/mb-icon/down_arrow_red@2x.png 2x, /assets/components/elements/mb-icon/down_arrow_red@3x.png 3x"
          />
        </div>
        <div
          class="cmMCategoryDropdown_Wrapper"
        >
          <ul
            class="cmMCategoryDropdown"
          >
            <li>
              <a
                class="cmMCategoryDropdown_Element"
                href="/store/category/wine"
              >
                <span
                  class="elMBText elMBText_ResetSpacing"
                >
                  wine
                </span>
              </a>
            </li>
            <li>
              <a
                class="cmMCategoryDropdown_Element"
                href="/store/category/liquor"
              >
                <span
                  class="elMBText elMBText_ResetSpacing"
                >
                  liquor
                </span>
              </a>
            </li>
            <li>
              <a
                class="cmMCategoryDropdown_Element"
                href="/store/category/beer"
              >
                <span
                  class="elMBText elMBText_ResetSpacing"
                >
                  beer
                </span>
              </a>
            </li>
            <li>
              <a
                class="cmMCategoryDropdown_Element"
                href="/store/category/mixers"
              >
                <span
                  class="elMBText elMBText_ResetSpacing"
                >
                  mixers
                </span>
              </a>
            </li>
            <li>
              <a
                class="cmMCategoryDropdown_Element"
                href="/store/promos/gift-guide"
              >
                <span
                  class="elMBText elMBText_ResetSpacing"
                >
                  gifts
                </span>
              </a>
            </li>
          </ul>
        </div>
      </div>
      <div
        class="cmSearch__ClickOutsideTarget"
      >
        <div>
          <div>
            <div
              class="elMBDynamicIcon cmMBrowseBar_SearchIcon"
            >
              <svg
                class=""
                height="29"
                viewBox="0 0 32 32"
                width="29"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  fill="#222"
                  fill-rule="nonzero"
                >
                  <path
                    d="M14.81 24.342c-5.145 0-9.31-4.22-9.31-9.42C5.5 9.72 9.665 5.5 14.81 5.5c5.145 0 9.31 4.22 9.31 9.421 0 5.2-4.165 9.421-9.31 9.421zm0-2c4.035 0 7.31-3.32 7.31-7.42 0-4.102-3.275-7.422-7.31-7.422-4.034 0-7.31 3.32-7.31 7.421 0 4.102 3.276 7.421 7.31 7.421z"
                  />
                  <path
                    d="M20.251 14.921h-2c0-1.936-1.543-3.5-3.44-3.5-.833 0-1.617.3-2.237.84l-1.313-1.509a5.388 5.388 0 0 1 3.55-1.33c3.007 0 5.44 2.464 5.44 5.5zM20.331 21.94l1.424-1.406 4.499 4.56-1.424 1.404z"
                  />
                </g>
              </svg>
            </div>
            <input
              class="elMBInput"
              placeholder="Search products"
              value=""
            />
          </div>
          <div
            class="el-touchable cmMBrowseBar_ClearSearch cmMBrowseBar_ClearSearch__SearchEmpty cmMBrowseBar_ClearSearch__SearchActive"
            role="button"
            tabindex="-1"
          >
            <img
              alt="clear"
              class="el-mbicon--clear"
              src="/assets/components/elements/mb-icon/clear.png"
              srcset="/assets/components/elements/mb-icon/clear@2x.png 2x, /assets/components/elements/mb-icon/clear@3x.png 3x"
            />
          </div>
          <div
            class="el-touchable"
            role="button"
            tabindex="-1"
          >
            <span
              class="elMBText elMBText_ResetSpacing cmMBrowseBar_CancelSearch"
            >
              Cancel
            </span>
          </div>
        </div>
        <div
          class="cmAutocompleteDropdown"
        />
      </div>
    </nav>
  </div>
  <div>
    <div
      class="elMBDrawer_Overlay elMBDrawer_Overlay__Hidden"
    />
    <div
      class="elMBDrawer elMBDrawer__Hidden"
    >
      <div>
        <div
          class="el-touchable"
          role="button"
          tabindex="-1"
        >
          <div
            class="elMBDynamicIcon"
          >
            <svg
              class=""
              height="30"
              viewBox="0 0 32 32"
              width="30"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14.62 16.251L7.656 9.289 9.07 7.875l6.962 6.962 6.963-6.962 1.414 1.414-6.962 6.962 6.962 6.962-1.414 1.414-6.963-6.962-6.962 6.962-1.414-1.414 6.962-6.962z"
                fill="#222"
                fill-rule="nonzero"
              />
            </svg>
          </div>
        </div>
        <div
          class="el-touchable"
          role="button"
          tabindex="-1"
        >
          <div>
            <img
              alt="on_demand"
              class="cmDeliveryMethodIcon_Multi"
              height="24"
              src="/assets/components/compounds/delivery_method_icon/on_demand.svg"
              width="24"
            />
            <div>
              <span
                class="elMBText elMBText_ResetSpacing"
              >
                Delivery to
              </span>
              <br />
              <span
                class="elMBText elMBText_ResetSpacing"
              >
                Add delivery address
              </span>
            </div>
          </div>
          <div
            class="elMBDynamicIcon"
          >
            <svg
              class=""
              height="28"
              viewBox="0 0 32 32"
              width="28"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14.297 10.922l.781-.625L19.64 16l-4.562 5.703-.78-.625L18.36 16z"
                fill="#494849"
                fill-rule="nonzero"
              />
            </svg>
          </div>
        </div>
        <ul>
          <li>
            <a
              class="elMBText_A elMBText elMBText_ResetSpacing"
              href="/store/"
            >
              Shop
            </a>
          </li>
          <li>
            <a
              class="elMBText_A elMBText elMBText_ResetSpacing"
              href="/referrals"
            >
              Share & Save $10
            </a>
          </li>
          <li>
            <a
              class="elMBText_A elMBText elMBText_ResetSpacing"
              href="/account/orders"
            >
              Orders
            </a>
          </li>
          <li>
            <a
              class="elMBText_A elMBText elMBText_ResetSpacing"
              href="/account/overview"
            >
              Account
              <span
                class="elMBText elMBText_ResetSpacing"
              >
                â€ƒ
              </span>
            </a>
          </li>
          <li>
            <div
              class="el-touchable"
              role="button"
              tabindex="-1"
            >
              <span
                class="elMBText elMBText_ResetSpacing"
              >
                Help
              </span>
            </div>
          </li>
        </ul>
        <div
          class="elMBAppStoreLink_Container"
        >
          <a
            href="https://itunes.apple.com/us/app/minibar-delivery-wine-liquor/id720850888?mt=8&uo=4&at=11l5XB&ct=minibar"
            id="app-store-link"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="Download on the App Store"
              src="/assets/components/elements/mb-app-store-link/ios-app-store.png"
              srcset="/assets/components/elements/mb-app-store-link/ios-app-store@2x.png 2x, /assets/components/elements/mb-app-store-link/ios-app-store@3x.png 3x"
            />
          </a>
          <a
            href="https://play.google.com/store/apps/details?id=minibar.android"
            id="play-store-link"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="Get it on Google Play"
              src="/assets/components/elements/mb-app-store-link/play-store.png"
              srcset="/assets/components/elements/mb-app-store-link/play-store@2x.png 2x, /assets/components/elements/mb-app-store-link/play-store@3x.png 3x"
            />
          </a>
        </div>
      </div>
    </div>
  </div>
</header>
`;
